src/components/KFileUpload/KFileUpload.vue

```diff
@@ -2,100 +2,76 @@
-      type="file"
-      @change="onFileChange"
-    />
-
-    <KIcon
-      v-if="type === 'image'"
-      class="image-upload-icon"
-      :color="iconColor"
-      :icon="icon"
-      :size="iconSize"
-      @click.prevent="updateFile"
-    />
-
-    <a
-      v-if="type === 'image'"
-      class="image-upload-description"
-      href="#"
-      @click.prevent="updateFile"
-    >
-      {{ fileValue ? fileValue : placeholder }}
-    </a>
-    <KButton
-      v-if="fileValue && removable"
-      appearance="primary"
-      class="remove-button"
-      :class="[label ? 'k-file-upload-btn-with-label' : 'k-file-upload-btn-without-label', { 'move-btn-right': type !== 'file' }]"
-      data-testid="remove-button"
-      size="small"
-      type="reset"
-      @click="resetInput"
-      @keyup.enter="resetInput"
-    >
-      <template #icon>
-        <KIcon
-          icon="close"
-          size="16"
-        />
+      <template
+        v-if="hasLabelTooltip"
+        #tooltip
+      >
+        <slot name="label-tooltip" />
       </template>
-    </KButton>
-    <KButton
-      v-if="type === 'file'"
-      :appearance="buttonAppearance"
-      class="k-file-upload-btn"
-      :class="[label ? 'k-file-upload-btn-with-label' : 'k-file-upload-btn-without-label']"
-      data-testid="k-file-upload-button"
-      size="small"
-      @click="updateFile"
-      @keyup.enter="updateFile"
-    >
-      {{ buttonText }}
-    </KButton>
-    <a
-      v-if="type === 'file'"
-      class="display-name"
-      :class="[label ? 'has-label' : 'has-no-label']"
-      href="#"
-      @click="updateFile"
-      @keyup.enter="updateFile"
-    >
-      {{ fileValue ? fileValue : placeholder }}
-    </a>
+    </KLabel>
+
+    <div class="file-upload-input-wrapper">
+      <span
+        :key="fileInputKey"
+        class="file-upload-input-text"
+        :class="{ 'placeholder': !fileValue, 'has-icon': $slots.icon, 'disabled': disabled }"
+      >
+        {{ fileValue ? fileValue : placeholder }}
+      </span>
+
+      <KInput
+        :id="inputId"
+        :key="fileInputKey"
+        :accept="accept"
+        class="upload-input"
+        :disabled="disabled"
+        :error="hasUploadError || error"
+        :error-message="errorMessage || fileSizeErrorMessage"
+        :help="help"
+        :max-file-size="maximumFileSize"
+        :placeholder="placeholder"
+        type="file"
+        @change="onFileChange"
+      >
+        <template
+          v-if="$slots.icon"
+          #before
+        >
+          <slot name="icon" />
+        </template>
+
+        <template #after>
+          <KButton
+            appearance="tertiary"
+            class="file-upload-button"
+            data-testid="file-upload-button"
+            :disabled="disabled"
+            size="small"
+            @click="updateFile"
+            @keyup.enter="updateFile"
+          >
+            {{ fileValue ? 'Clear' : buttonText }}
+          </KButton>
```

Comment:
issue: clicking the button does not work; I have to click on the input to trigger the upload

Reply:
It seems to work on the sandbox site, but not on the docs site 🤔

Reply:
Yes and it works as expected in the docs hosted locally - but not on preview... no big changes in that piece of functionality though so I'm not sure what's wrong (it works on Kongponents main though) 🤔

Reply:
Fixed, needed `ClientOnly` in docs